File: задание 3/main.swift
import Foundation

func initList() -> [Int] {
    while true {
        do {
            guard let input = readLine() else {
                print("Error: No input found! Try again: ")
                continue
            }
            let nums = try input.split(separator: " ").map { str in
                guard let num = Int(str) else {
                    throw InputError.invalidValue
                }
                return num
            }
            return nums
        } catch InputError.invalidValue {
            print("Error: An invalid value has been entered! Try again: ")
        } catch {
            print("An unknown error occurred!")
        }
    }
}

enum InputError: Error {
    case invalidValue
}

func isPrime(_ num: Int) -> Bool {
    if num <= 1 {
        return false
    }
    for i in 2..<Int(Double(num).squareRoot()) + 1 {
        if num % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    let numbers = initList()

    let primeCount = numbers.filter { isPrime($0) }.count

    print("Number of prime numbers: \(primeCount)")
}

main()

File: задание 3/main.rb
def init_list
    loop do
      begin
        input = gets
        return [] if input.nil? # Проверяем на nil
        nums = input.chomp.split.map(&:to_i)
        return nums
      rescue ArgumentError
        puts "Error: An invalid value has been entered! Try again: "
      end
    end
  end
  
  
  def is_prime(num)
    return false if num <= 1
  
    (2..Math.sqrt(num).to_i).each do |i|
      return false if num % i == 0
    end
  
    true
  end
  
  def main
    numbers = init_list
  
    prime_count = numbers.count { |num| is_prime(num) }
  
    puts "Number of prime numbers: #{prime_count}"
  end
  
  if __FILE__ == $PROGRAM_NAME
    main
  end